opencti-platform/opencti-front/src/private/components/observations/stix_cyber_observables/StixCyberObservableCreation.jsx

```diff
@@ -419,16 +646,101 @@ const StixCyberObservableCreation = ({
+                    .when(['hashes_MD5', 'hashes_SHA-1', 'hashes_SHA-256', 'hashes_SHA-512'], {
+                      is: (a, b, c, d) => !a && !b && !c && !d,
+                      then: () => Yup.string().required(t_i18n('MD5, SHA-1, SHA-256, SHA-512, or name is required')),
+                    }),
+                };
+
+                requiredOneOfFields = [
+                  ['hashes_MD5', 'hashes_SHA-1'],
+                  ['hashes_MD5', 'hashes_SHA-256'],
+                  ['hashes_MD5', 'hashes_SHA-512'],
+                  ['hashes_MD5', 'name'],
+                  // ['hashes_SHA-1', 'hashes_MD5'],
+                  ['hashes_SHA-1', 'hashes_SHA-256'],
+                  ['hashes_SHA-1', 'hashes_SHA-512'],
+                  ['hashes_SHA-1', 'name'],
+                  // ['hashes_SHA-256', 'hashes_MD5'],
+                  // ['hashes_SHA-256', 'hashes_SHA-1'],
+                  ['hashes_SHA-256', 'hashes_SHA-512'],
+                  ['hashes_SHA-256', 'name'],
+                  // ['hashes_SHA-512', 'hashes_MD5'],
+                  // ['hashes_SHA-512', 'hashes_SHA-1'],
+                  // ['hashes_SHA-512', 'hashes_SHA-256']
+                  ['hashes_SHA-512', 'name'],
+                ];
               } else if (attribute.value === 'value') {
                 initialValues[attribute.value] = inputValue || '';
+                // Dynamically include value field for Singular Observable type Object form validation
+                const exceededMessage = t_i18n('You have exceeded the max number of values.');
+                extraFieldsToValidate = {
+                  [attribute.value]: Yup
+                    .string()
+                    .when(['bulk_value_field'], {
+                      is: (a) => !a,
+                      then: () => Yup.string().required(t_i18n('A value is required')),
+                    }),
+                  bulk_value_field: Yup
+                    .string()
+                    .when([attribute.value], {
+                      is: (a) => !a,
+                      then: () => Yup.string().required(t_i18n('Multiple value entry is required or Cancel this form')).test('len', exceededMessage, (val) => val.split('\n').length < 51),
+                    }),
+                };
+                requiredOneOfFields = [
+                  [attribute.value, 'bulk_value_field'],
+                ];
               } else {
                 initialValues[attribute.value] = '';
               }
             }
+            const stixCyberObservableValidationBaseFields = {
+              x_opencti_score: Yup.number().nullable(),
+              x_opencti_description: Yup.string().nullable(),
+              createIndicator: Yup.boolean(),
+            };
+            const stixCyberObservableValidationFinal = (extraRequiredFields = null) => Yup.object().shape({
+              ...stixCyberObservableValidationBaseFields,
+              ...extraRequiredFields,
+            }, requiredOneOfFields);
```

Comment:
no need to use a function here, all the info needed is already there when creating the variable `stixCyberObservableValidationBaseFields` so we can build the full validation object directly